src/features/session_trace/aggregate/index.js

```diff
@@ -106,6 +114,58 @@ export class Aggregate extends FeatureBase {
     drain(this.agentIdentifier, this.featureName)
   }
 
+  startTracing (startupBuffer, switchToMode = MODE.FULL) {
+    this.mode = MODE.FULL // TO DO: create error mode for this feature
+    if (typeof PerformanceNavigationTiming !== 'undefined') {
+      this.storeTiming(window.performance.getEntriesByType('navigation')[0])
+    } else {
+      this.storeTiming(window.performance.timing)
+    }
+
+    const scheduler = new HarvestScheduler('resources', {
+      onFinished: onHarvestFinished.bind(this),
+      retryDelay: this.harvestTimeSeconds
+    }, this)
+    scheduler.harvest.on('resources', prepareHarvest.bind(this))
+    scheduler.runHarvest({ needResponse: true }) // sends first stn harvest immediately
+    startupBuffer.decide(true) // signal to flush data that was pending RUM response flag for the next harvest
+
+    function onHarvestFinished (result) {
+      if (result.sent && result.responseText && !this.ptid) { // continue interval harvest only if ptid was returned by server on the first
+        this.agentRuntime.ptid = this.ptid = result.responseText
+        scheduler.startTimer(this.harvestTimeSeconds)
+      }
+
+      if (result.sent && result.retry && this.sentTrace) { // merge previous trace back into buffer to retry for next harvest
+        Object.entries(this.sentTrace).forEach(([name, listOfSTNodes]) => {
+          if (this.nodeCount >= this.maxNodesPerHarvest) return
+
+          this.nodeCount += listOfSTNodes.length
+          this.trace[name] = this.trace[name] ? listOfSTNodes.concat(this.trace[name]) : listOfSTNodes
+        })
+        this.sentTrace = null
+      }
+    }
+    function prepareHarvest (options) {
+      /* Standalone refers to the legacy version of ST before the idea of 'session' or the Replay feature existed.
+        It has a different behavior on returning a payload for harvest than when used in tandem with either of those concepts. */
+      if (this.isStandalone) {
+        if (now() > MAX_TRACE_DURATION) { // been collecting for over the longest duration we should run for, empty trace object so ST has nothing to send
+          scheduler.stopTimer()
+          this.trace = {}
+          return
+        }
+        // Only harvest when more than some threshold of nodes are pending, after the very first harvest.
+        if (this.ptid && this.nodeCount <= REQ_THRESHOLD_TO_SEND) return
+      } else {
+        // With sessions on harvest intervals, visible pages will send payload regardless of pending nodes but backgrounded pages will still abide by threshold.
+        if (this.ptid && document.visibilityState === 'hidden' && this.nodeCount <= REQ_THRESHOLD_TO_SEND) return
```

Comment:
Lets not limit the background interval by nodeCount for now, to ensure we always send all data when a replay is in progress.

This might change later